src/views/TopologyDisplay.vue

```diff
@@ -0,0 +1,340 @@
+  open: string[] = []
+  columnItems: ColumnItem[] = []
+  tabs: TopologyDisplayTab[] = []
+  baseUrl!: string
+  webServiceProvider: PiWebserviceProvider = {} as PiWebserviceProvider;
+  urlTopologyNodeMap: Map<string, string> = new Map<string, string>()
+  displays: DisplayConfig[] = []
+  selectedTab = null;
+  allDisplays: DisplayConfig[][] = []
+  requests: ActionRequest[][] = [];
+  plots: string[] = [];
+  selectedItem: number = -1
+  wmsLayerId: string | undefined = ''
+  filterIds: string[] | undefined = []
+  topologyNode: TopologyNode | undefined = undefined
+  showMap = true
+  showTimeSeries = false
+  topologyMap: Map<string, TopologyNode> = new Map<string, TopologyNode>();
+
+
+  created(): void {
+    this.baseUrl = this.$config.get('VUE_APP_FEWS_WEBSERVICES_URL')
+  }
+
+  async mounted(): Promise<void> {
+    const transformRequestFn = this.getTransformRequest()
+    this.webServiceProvider = new PiWebserviceProvider(this.baseUrl, {transformRequestFn});
+    let nodes: TopologyNodeResponse = await this.webServiceProvider.getTopologyNodes();
+    this.createTopologyMap(nodes.topologyNodes, this.topologyMap)
+    this.columnItems = await this.loadNodes(nodes)
+    this.open = [this.columnItems[0].id]
+    this.tabs = this.getTopologyTabs(nodes)
+
+  }
+
+
+  async updateLocations(locationIds: string[]) {
+    if (this.filterIds === undefined) return
+    if (locationIds.length === 0) {
+      if (this.topologyNode?.displayGroups !== undefined) {
+        await this.showDisplayGroupsTimeSeries()
+      } else {
+        this.showTimeSeries = false;
+      }
+      return
+    }
+    const [displays, requests] = await fetchTimeSeriesDisplaysAndRequests(this.webServiceProvider, this.filterIds, locationIds)
+    this.displays = displays
+    this.plots = []
+    await this.updateTimeSeries(requests, {startTime: this.startTime, endTime: this.endTime, thinning: true})
+    this.showTimeSeries = true
+  }
+
+  @Watch('selectedItem')
+  async onPlotChanged(): Promise<void> {
+    this.displays = this.allDisplays[this.selectedItem];
+    await this.loadTimeSeries(this.selectedItem);
+  }
+
+  @Watch('nodeId')
+  async onNodeChange(): Promise<void> {
+
+    const topologyNode: TopologyNode | undefined = this.topologyMap.get(this.nodeId)
+    if (topologyNode === undefined) return
+    this.showMap = topologyNode?.gridDisplaySelection?.plotId !== undefined || topologyNode.filterIds !== undefined
+    this.showTimeSeries = topologyNode?.displayGroups !== undefined
+
+    this.$nextTick(function () {
+      this.topologyNode = topologyNode //this is needed otherwise the property topologyNode is not passed to the explorer component when showMap was false
+      this.filterIds = topologyNode?.filterIds
+      this.wmsLayerId = topologyNode?.gridDisplaySelection?.plotId
+
+
+      if (!this.showTimeSeries) return
+      this.showDisplayGroupsTimeSeries();
+    })
+
+
+  }
+
+  private async showDisplayGroupsTimeSeries() {
```

Comment:
Moet deze functie misschien iets van `fetch...` heten?

Return type annotation is ook wel prettig qua documentatie.